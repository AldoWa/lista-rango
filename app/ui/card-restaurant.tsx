'use client';

import Link from "next/link";
import VeganImage from "@/public/vegan-restaurant.png";
import Image from "next/image";
import { useState } from "react";

interface CardRestaurantProps {
  name: string;
  address: string;
  isOpen: boolean;
  photoSrc: string;
}

export function CardRestaurant({
  name,
  address,
  isOpen,
  photoSrc,
}: CardRestaurantProps) {
  const [finalSrc, setFinalSrc] = useState('');

  function changeLoadStatus(e: HTMLImageElement){
    setFinalSrc(e.src)
  }

  return (
    <Link
      href="/"
      aria-label="Go to the restaurant"
      className="shadow-md flex items-center 
        rounded
        relative	
        h-28
      "
    >
      <Image
        src={finalSrc ? photoSrc : VeganImage}
        alt="Restaurant Logo"
        width={112}
        height={112}
        className="h-28"
        onLoad={e => changeLoadStatus(e.target as HTMLImageElement)}
        quality={100}
      />

      <div className="ml-6">
        <p className="text-base">{name}</p>
        <p className="text-xs">{address}</p>
      </div>

      <div
        aria-live="polite"
        role="status"
        className="bg-purple-900 rounded-full
          w-14
          h-14
          flex
          items-center
          justify-center
          absolute
          -top-2 -right-4
        "
      >
        <span className="text-white text-xs leading-3 text-center font-bold ">
          {isOpen ? "Aberto agora" : "Fechado"}
        </span>
      </div>
    </Link>
  );
}