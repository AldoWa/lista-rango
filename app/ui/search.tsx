'use client'

import { usePathname, useSearchParams, useRouter } from "next/navigation"

export function Search(){
  const searchParams = useSearchParams()
  const pathName = usePathname()
  const { replace } = useRouter();

  function handleOnSearch(term: string) {
    const params = new URLSearchParams(searchParams)
    if(term) {
      params.set('search', term)
    } else{
      params.delete('search')
    } 
    replace(`${pathName}?${params.toString()}`)
  }
  
  return (
    <div data-testid="home-form" className='flex justify-center'>
      <input 
        type="text" 
        className='
          shadow-md px-8
          w-full 
          rounded-full 
          bg-slate-100 h-10 
          focus:outline-none
          my-8
          max-w-4xl
        '
        onChange={(e) => {
          handleOnSearch(e.target.value)
        }}
        defaultValue={searchParams.get('query')?.toString()}
        placeholder='Buscar estabelecimento'
      />
    </div>
  )
}