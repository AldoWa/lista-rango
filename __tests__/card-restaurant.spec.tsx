import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { CardRestaurant } from '@/app/ui/card-restaurant';

describe('card-restaurant', () => {
  it('Should render the card restaurant component with isOpen true', async () => {
    render(<CardRestaurant 
      address='Address 1'
      name='Restaurant 1'
      isOpen={true}
      photoSrc='https://via.placeholder.com/112x112'
    />)
    
    const restaurantContainer = screen.getByTestId('restaurant-card')
    const restaurantName = screen.getByTestId('restaurant-name')
    const restaurantAddress = screen.getByTestId('restaurant-address')
    const restaurantOpened = screen.getByTestId('restaurant-opened')


    expect(restaurantContainer).toBeInTheDocument()
    expect(restaurantName).toBeInTheDocument()
    expect(restaurantAddress).toBeInTheDocument()
    expect(restaurantOpened).toBeInTheDocument()

    expect(restaurantOpened).toHaveTextContent('Aberto agora')
    expect(restaurantOpened).toHaveClass('text-white')
  });

  it('Should render the card restaurant component with isOpen false', async () => {
    render(<CardRestaurant 
      address='Address 1'
      name='Restaurant 1'
      isOpen={false}
      photoSrc='https://via.placeholder.com/112x112'
    />)
    
    const restaurantOpened = screen.getByTestId('restaurant-opened')

    const image = screen.getByRole('img')
    expect(image).toBeInTheDocument()

    expect(restaurantOpened).toBeInTheDocument()
    expect(restaurantOpened).toHaveTextContent('Fechado')
  })

  it('Should match snapshot', async () => {
    const { container } = render(<CardRestaurant 
      address='Address 1'
      name='Restaurant 1'
      isOpen={false}
      photoSrc='https://via.placeholder.com/112x112'
    />)
    expect(container).toMatchSnapshot()
  })
})  