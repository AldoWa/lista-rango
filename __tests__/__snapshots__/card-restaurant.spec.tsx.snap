// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card-restaurant Should match snapshot 1`] = `
<div>
  <a
    aria-label="Go to the restaurant"
    class="shadow-md flex items-center  rounded relative  h-28 "
    data-testid="restaurant-card"
    href="/restaurants/1/menu"
  >
    <img
      alt="Restaurant Logo"
      class="h-28"
      data-nimg="1"
      decoding="async"
      height="112"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=256&q=100"
      srcset="/_next/image?url=%2Fimg.jpg&w=128&q=100 1x, /_next/image?url=%2Fimg.jpg&w=256&q=100 2x"
      style="color: transparent;"
      width="112"
    />
    <div
      class="ml-6"
    >
      <p
        class="text-base"
        data-testid="restaurant-name"
      >
        Restaurant 1
      </p>
      <p
        class="text-xs"
        data-testid="restaurant-address"
      >
        Address 1
      </p>
    </div>
    <div
      aria-live="polite"
      class="bg-purple-900 rounded-full
          w-14
          h-14
          flex
          items-center
          justify-center
          absolute
          -top-2 -right-4
          bg-[#B5ABD4]
        "
      role="status"
    >
      <span
        class="text-white text-xs leading-3 text-center font-bold"
        data-testid="restaurant-opened"
        role="alert"
      >
        Fechado
      </span>
    </div>
  </a>
</div>
`;
